x-pack/test/api_integration/apis/ml/results/get_partition_fields_values.ts

```diff
@@ -0,0 +1,247 @@
+      model_plot_config: { enabled: enableModelPlot },
+    } as Job;
+  }
+
+  function getDatafeedConfig(jobId: string) {
+    return {
+      datafeed_id: `datafeed-${jobId}`,
+      indices: ['ft_farequote'],
+      job_id: jobId,
+      query: { bool: { must: [{ match_all: {} }] } },
+    } as Datafeed;
+  }
+
+  async function createMockJobs() {
+    await ml.api.createAndRunAnomalyDetectionLookbackJob(
+      getJobConfig('fq_multi_1_ae'),
+      getDatafeedConfig('fq_multi_1_ae')
+    );
+
+    await ml.api.createAndRunAnomalyDetectionLookbackJob(
+      getJobConfig('fq_multi_2_ae', false),
+      getDatafeedConfig('fq_multi_2_ae')
+    );
+  }
+
+  describe('GetAnomaliesTableData', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await createMockJobs();
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should fetch anomalous only field values within the time range with an empty search term sorting by anomaly score', async () => {
+      const requestBody = {
+        jobId: 'fq_multi_1_ae',
+        criteriaFields: [{ fieldName: 'detector_index', fieldValue: 0 }],
+        earliestMs: 1454889600000, // February 8, 2016 12:00:00 AM GMT
+        latestMs: 1454976000000, // February 9, 2016 12:00:00 AM GMT,
+        searchTerm: {},
+        fieldsConfig: {
+          partition_field: {
+            applyTimeRange: true,
+            anomalousOnly: true,
+            sort: { by: 'anomaly_score', order: 'desc' },
+          },
+        },
+      };
+
+      const { body, status } = await supertest
+        .post(`/api/ml/results/partition_fields_values`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(requestBody);
+      ml.api.assertResponseStatusCode(200, status, body);
+
+      expect(body.partition_field.name).to.eql('airline');
+      expect(body.partition_field.values.length).to.eql(6);
+      expect(body.partition_field.values[0].value).to.eql('ACA');
+      expect(body.partition_field.values[0].maxRecordScore).to.be.above(50);
+      expect(body.partition_field.values[1].value).to.eql('JBU');
+      expect(body.partition_field.values[1].maxRecordScore).to.be.above(30);
+      expect(body.partition_field.values[2].value).to.eql('SWR');
+      expect(body.partition_field.values[2].maxRecordScore).to.be.above(25);
+      expect(body.partition_field.values[3].value).to.eql('BAW');
+      expect(body.partition_field.values[3].maxRecordScore).to.be.above(10);
+      expect(body.partition_field.values[4].value).to.eql('TRS');
+      expect(body.partition_field.values[4].maxRecordScore).to.be.above(7);
+      expect(body.partition_field.values[5].value).to.eql('EGF');
+      expect(body.partition_field.values[5].maxRecordScore).to.be.above(2);
```

Comment:
Seeing that the actual scores are close to these bounds (51.1, 31.0, 25.9, 10.9, 8.3, 3.0), I think there's a risk of future failures when the backend delivers slightly different values. In most other places of Kibana tests, we tried to avoid result value assertions (quality of results is tested separately on the QA side). So I think we could do one of the following:
1. Keep the `maxRecordScore` assertions but give it more tolerance (e.g. go down by ~ 5)
2. Modify the `maxRecordScore` assertions to just check `> 0`
3. Remove the `maxRecordScore` assertions.

My personal vote goes to number 2.